docs/knt-backend-doc.yml

```diff
@@ -0,0 +1,363 @@
+openapi: '3.0.2'
+info:
+  title: KnT Backend
+  version: '1.0'
+servers:
+  - url: http://127.0.0.1:5000/
+paths:
+  /ping:
+    get:
+      description: "Basic ping function accessable without any keys"
+      tags:
+        - "Misc"
+      responses:
+        '200':
+          description: OK
+  /users:
+    get:
+      tags:
+        - "User Basic"
+      security:
+        - ApiKeyAuth: []
+      description: 'Get basic information about all registered users accessible with a user api key'
+      parameters: []
+      responses:
+        '200':
+          description: A list of basic user objects with removed sensitive information
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/user"
+                  
+        '401':
+          $ref: "#/components/responses/UnauthorizedError"
+        '407':
+          $ref: "#/components/responses/ForbidenError"
+  /products:
+    get:
+      tags:
+        - "Product Basic"
+      description: 'Retrieve a list of products from the database'
+      parameters: []
+      security:
+        - ApiKeyAuth: []
+      responses:
+        '200':
+          description: A list of product objects
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/product"
+                  
+        '401':
+          $ref: "#/components/responses/UnauthorizedError"
+        '407':
+          $ref: "#/components/responses/ForbidenError"
+  /admin/products:
+    post:
+      tags:
+        - "Product Admin"
+      description: 'Create a new product in the database'
+      security:
+        - ApiKeyAuth: []
+      requestBody:
+        description: 'Basic product to create'
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/product'
+
+      responses:
+        '201':
+          description: "Created"
+        '401':
+          $ref: "#/components/responses/UnauthorizedError"
+        '407':
+          $ref: "#/components/responses/ForbidenError"
+  /users/{userId}/purchase:
+    post:
+      tags:
+        - "User Basic"
+      description: 'Send purchase ammounts'
```

Comment:
Description is a bit too short, maybe something like "Send the list of items, specified by ID with the amount indicated, that the current user wants to purchase." (Nothing dire, just a suggestion)